hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceProxy.java

```diff
@@ -68,7 +68,7 @@
  * </li>
  * </ol>
  */
-public final class HazelcastInstanceProxy implements HazelcastInstance, SerializationServiceSupport {
+public class HazelcastInstanceProxy implements HazelcastInstance, SerializationServiceSupport {
     protected volatile HazelcastInstanceImpl original;
```

Comment:
yes. that's why I'm quite reluctant to open it for extensions.
we are in a bad position due having too open internal classes. and now we are opening them even more. 

I'm willing to accept it in the cases where it's necessary. but then I want to see a good reasoning behind it.